fix: remove hardcoded path from test_makefile.sh

## Problem

CI pipeline failing on GitHub Actions due to hardcoded absolute path in test script.

## Error

```
tests/test_makefile.sh: line 9: cd: /home/puupa/Projects/Poker/HandEvaluation_C: No such file or directory
make: *** [Makefile:64: test] Error 1
```

## Root Cause

`tests/test_makefile.sh` contains a hardcoded path to the local development machine:
- Local path: `/home/puupa/Projects/Poker/HandEvaluation_C`
- GitHub Actions path: `/home/runner/work/poker-hand-evaluation-c/poker-hand-evaluation-c`

The script fails because this absolute path doesn't exist on CI runners.

## Solution

Remove hardcoded absolute path and use relative paths or environment variables instead.

**Options:**
1. Use `${GITHUB_WORKSPACE}` environment variable on CI
2. Use relative paths from test script location
3. Remove the `cd` command entirely if not needed

## Acceptance Criteria

- [ ] Test script runs successfully on GitHub Actions
- [ ] Test script still works locally
- [ ] No hardcoded absolute paths in any test scripts
- [ ] CI pipeline passes

## Impact

**Severity:** High - Blocking CI pipeline
**Affected Jobs:** Test suite, CI workflow

## Files to Modify

- `tests/test_makefile.sh` (line 9)

## Related

- Failing CI run: https://github.com/arthurvanacker/poker-hand-evaluation-c/actions/runs/18236050888

--- a/tests/test_makefile.sh
+++ b/tests/test_makefile.sh
@@ -5,7 +5,9 @@
 
 set -e
 
-PROJECT_ROOT="/home/puupa/Projects/Poker/HandEvaluation_C"
+# Get project root directory (parent of tests/)
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "$SCRIPT_DIR/.." && pwd)"
 cd "$PROJECT_ROOT"
 
 echo "=========================================="

--- a/tests/test_script_portability.sh
+++ b/tests/test_script_portability.sh
@@ -1,0 +1,54 @@
+#!/bin/bash
+
+# Test script portability - verifies test_makefile.sh doesn't use hardcoded paths
+# This test ensures the script works in any environment (local, CI, etc.)
+
+set -e
+
+echo "=========================================="
+echo "Testing Script Portability"
+echo "=========================================="
+echo ""
+
+# Test 1: Verify no hardcoded absolute paths in script
+echo "[TEST 1] Checking for hardcoded absolute paths..."
+if grep -q "PROJECT_ROOT=\"/home/" tests/test_makefile.sh; then
+    echo "FAIL: Found hardcoded absolute path in test_makefile.sh"
+    grep "PROJECT_ROOT=" tests/test_makefile.sh
+    exit 1
+fi
+echo "PASS: No hardcoded absolute paths found"
+echo ""
+
+# Test 2: Verify script uses relative path logic
+echo "[TEST 2] Verifying script uses relative paths..."
+if ! grep -q 'cd.*dirname.*BASH_SOURCE' tests/test_makefile.sh; then
+    echo "FAIL: Script doesn't use BASH_SOURCE for relative path resolution"
+    exit 1
+fi
+echo "PASS: Script uses relative path resolution"
+echo ""
+
+# Test 3: Run from tests directory
+echo "[TEST 3] Running test_makefile.sh from tests/ directory..."
+cd tests || exit 1
+if ! ./test_makefile.sh > /dev/null 2>&1; then
+    echo "FAIL: Script failed when run from tests/ directory"
+    exit 1
+fi
+cd .. || exit 1
+echo "PASS: Script works from tests/ directory"
+echo ""
+
+# Test 4: Run from project root
+echo "[TEST 4] Running test_makefile.sh from project root..."
+if ! ./tests/test_makefile.sh > /dev/null 2>&1; then
+    echo "FAIL: Script failed when run from project root"
+    exit 1
+fi
+echo "PASS: Script works from project root"
+echo ""
+
+echo "=========================================="
+echo "ALL PORTABILITY TESTS PASSED!"
+echo "=========================================="

